pkg/kapp/diff/change_set.go

```diff
@@ -4,14 +4,23 @@
 package diff
 
 import (
+	"context"
 	"fmt"
 	"strings"
 
 	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
 )
 
+type ChangeSetMode string
+
+const (
+	ExactChangeSetMode              = "Exact"
+	AgainstLastAppliedChangeSetMode = "AgainstLastApplied"
+	ServerSideApplyChangeSetMode    = "ServerSideApply"
+)
+
 type ChangeSetOpts struct {
-	AgainstLastApplied bool
+	Mode ChangeSetMode
```

Comment:
after reading through flag "adjusting" above, i think we should just turns this into:

```
type ChangeSetOpts struct {
        ServerSideApply bool
	AgainstLastApplied bool
}
```

anywhere where we are doing a switch based on mode, we should just do:

```
switch {
case opts.ServerSideApply: ...
case opts.AgainstLastApplied: ...
default: ...
}
```

the reasoning for this, it seems that mode enum introduces quite a bit of gymanistics necessary to compute what it should be.